app/controllers/users_controller.rb

```diff
@@ -21,9 +22,25 @@ def create
     end
   end
 
+  def update
```

Comment:
Could be a possible refactor opportunity to eventually pull some of this logic into private methods.